### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by karlwestin**
Im sending this pull request to start a discussion
Unfortunately @quesebifurcan and i found a few bugs working on this. Those bugs are described here https://github.com/eHealthAfrica/kazana-raw-data/commit/d408140c7b12a278d827757cf7d4f8e28e7f5c04 (but that commit needs some discussion i think)
Another thing to discuss that i need help with: Can we use kazana-server instead of this very complicated mock that i have here: https://github.com/eHealthAfrica/kazana-raw-data/blob/chore/unit-tests/tests/utils/kazana-mock-factory.js
The issue there is that the kazana server would need to be started, stopped and re-populated between every test. 

Anyway hope someone has time to write a little feedback here

### CODE REVIEW

File: lib/routes/submissions/remove.js [hunk 1]
```diff
@@ -15,24 +15,26 @@ function removeSubmissionHandler (request, reply) {
...
+    promise = store.find(id)
+      .then(failIfNotByUser.bind({}, user))
   }
 
   server.log(['raw-data', id, 'silly'], 'Removing')
 
   // .remove does not yet support setting other properties
   // https://github.com/hoodiehq/pouchdb-hoodie-api/issues/48
-  promise.then(store.update.bind(null, id))
+  promise.then(store.update.bind({}, id, deleted))
```
Comment (gr2m) [inline lib/routes/submissions/remove.js]: just wondering: any reason to use `{}` over `null`?

File: package.json [hunk 2]
```diff
@@ -32,9 +33,17 @@
     "lodash": "3.10.1"
   },
   "devDependencies": {
-    "kazana": "6.3.5",
```
Comment (gr2m) [inline package.json]: Did you remove `kazana` on purpose? It's needed to start the dev server with `npm start`

Comment (karl-ehealth) [inline package.json]: no that was probably a merge fuck up :)

File: lib/routes/submissions/remove.js (hunk 1 above)
Comment (karl-ehealth) [inline lib/routes/submissions/remove.js]: no just habit, can go w/ null instead

**Comment by gr2m** [comment]
I can't test locally because `git` seems to be disabled on my public wifi here, but travis says it's green so go for it, looks good to me :)

**Comment by patriciagarcia** [comment]
@gr2m and/or @jo, what's the best way to merge this? One of the PRs contains fixes for 4 different bugs: https://github.com/eHealthAfrica/kazana-raw-data/commit/cd045746c24d9174ed56d4b6e4c02a78b3a6c8e1. How should this be merged so that it works well with semantic release?

**Comment by gr2m** [comment]
Depending on how hard it is to split up. If it's super easy, I'd split it up. If not, I'd just keep it as is, and then manually update the github release to add the 4 bugs to the changelog.

**Comment by karlwestin** [comment]
so i rewrote the commit msg, 2 of the bug fixes where in the same fix, so there's only 3 fix commits
hope this is ok!

**Comment by jo** [comment]
Yes thats ok. We can update the gh release notes with the information, like @gr2m said.

File: package.json [hunk 3]
```diff
@@ -32,9 +33,17 @@
...
+    "boom": "^2.8.0",
     "bundle-bump-bot": "1.1.0",
+    "hapi": "10.4.0",
+    "kazana": "6.3.5",
```
Comment (jo) [inline package.json]: no, its still here :)

File: package.json [hunk 4]
```diff
@@ -32,9 +33,17 @@
     "lodash": "3.10.1"
   },
   "devDependencies": {
-    "kazana": "6.3.5",
+    "boom": "^2.8.0",
```
Comment (jo) [inline package.json]: do we need to declare it also as dev dependency?

Comment (gr2m) [inline package.json]: `boom` is already in `dependencies` and can be removed from `devDependencies`

**Comment by jo** [comment]
lgtm

**Pull request merged**